File: tree/0226_invert_binary_tree.js
/*
Given the root of a binary tree, invert the tree, and return its root.
*/

// https://leetcode.com/problems/invert-binary-tree/description/

var invertTree = function(root) {
    if (!root) return null;
    // invert left and right recursively
    let left = invertTree(root.left);
    let right = invertTree(root.right);
    // swap left and right
    root.left = right;
    root.right = left;
    return root;
};

File: tree/binary_tree/0226_invert_binary_tree.py
# Given the root of a binary tree, invert the tree, and return its root.

# https://leetcode.com/problems/invert-binary-tree/description/

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return None
        left = self.invertTree(root.left)
        right = self.invertTree(root.right)
        root.left = right
        root.right = left
        return root



